pkg/controller/handlers/poweruserworkspace/poweruserworkspace.go

```diff
@@ -172,10 +173,8 @@ func (h *Handler) cleanupWorkspaceForDemotionToPowerUser(ctx context.Context, cl
 		}
 
 		for _, acr := range acrs.Items {
-			if !acr.Spec.Generated {
-				if err := client.Delete(ctx, &acr); err != nil {
-					return err
-				}
+			if err := client.Delete(ctx, &acr); err != nil {
```

Comment:
I don't think we want to delete the generated access control rule completely. This will mean that all of the user's custom MCP servers will also get deleted because the access control rule is gone.

The handler that creates the generated access control rule makes this distinction between power user and power user plus: https://github.com/obot-platform/obot/blob/main/pkg/controller/handlers/poweruserworkspace/poweruserworkspace.go#L226-L233

I think we should just update the generated access control rule. Similarly, I think if the user it upgraded from power user to power user plus, then we should also make the corresponding change.

Reply:
Yeah 

> This will mean that all of the user's custom MCP servers will also get deleted because the access control rule is gone.

We made a change to ensure that the owner of the workspace will see their entries regardless of ACR for a better UX. So when PU user deletes all its ACR they should still see their entries.

>  Similarly, I think if the user it upgraded from power user to power user plus, then we should also make the corresponding change.

Yeah, we dicussed this a bit. I specifically didn't make it this case, because there could be a case where PU is upgraded to PUP and immediately their mcp server are shared by the world. So rather making the change automatically, when they got promoted to power user plus, they should be able to modify ACR to decide who to share.